apps/src/templates/census2017/CensusTeacherBanner.jsx

```diff
@@ -0,0 +1,368 @@
+    width: styleConstants['content-width'],
+    backgroundColor: color.white,
+    overflowWrap: 'break-word'
+  },
+  message: {
+    marginTop: 0,
+    marginBottom: 20,
+    marginLeft: 20,
+    marginRight: 20,
+  },
+  radio: {
+    verticalAlign: 'top',
+    marginRight: 10,
+  },
+  title: {
+    marginBottom: 0,
+  },
+  updateSchool: {
+    fontSize: '85%',
+    marginTop: 0,
+    marginBottom: 0,
+  },
+};
+
+export default class CensusTeacherBanner extends Component {
+  static propTypes = {
+    schoolYear: PropTypes.number.isRequired,
+    onSubmit: PropTypes.func.isRequired,
+    onDismiss: PropTypes.func.isRequired,
+    onPostpone: PropTypes.func.isRequired,
+    onChange: PropTypes.func.isRequired,
+    ncesSchoolId: PropTypes.string.isRequired,
+    question: PropTypes.oneOf(['how_many_10_hours', 'how_many_20_hours']).isRequired,
+    selection: PropTypes.bool,
+    teacherId: PropTypes.number.isRequired,
+    teacherName: PropTypes.string.isRequired,
+    teacherEmail: PropTypes.string.isRequired,
+    showUnknownError: PropTypes.bool,
+  };
+
+  componentDidMount() {
+    this.loadSchoolName(this.props.ncesSchoolId);
+  }
+
+  state = {
+    showSchoolInfoForm: false,
+    country: 'United States',
+    schoolType: '',
+    ncesSchoolId: null,
+    schoolName: '',
+    schoolDisplayName: null,
+    schoolState: '',
+    schoolZip: '',
+    schoolLocation: '',
+  };
+
+  handleCountryChange = (_, event) => {
+    const newCountry = event ? event.value : '';
+    this.setState({country: newCountry});
+  }
+
+  handleSchoolTypeChange = (event) => {
+    const newType = event ? event.target.value : '';
+    this.setState({schoolType: newType});
+  }
+
+  handleSchoolChange = (_, event) => {
+    const newSchool = event ? event.value : '';
+    this.setState({ncesSchoolId: newSchool});
+    this.loadSchoolName(newSchool);
+  }
+
+  handleSchoolNotFoundChange = (field, event) => {
+    const newValue = event ? event.target.value : '';
+    this.setState({
+      [field]: newValue
+    });
+  }
+
+  showSchoolInfoForm = () => {
+    this.setState({showSchoolInfoForm: true});
+  }
+
+  hideSchoolInfoForm = () => {
+    this.setState({
+      showSchoolInfoForm: false,
+      showSchoolInfoErrors: false,
+    });
+  }
+
+  dismissSchoolInfoForm = () => {
+    // This is handling the case where the user dismissed the
+    // "update your school" view, not the entire banner. In that case, there
+    // may have been partial state about the school changed. We do not want
+    // to use that partial state in the census submission so we need to reset
+    // to the previous values.
+    this.setState({
+      showSchoolInfoForm: false,
+      country: 'United States',
+      schoolType: '',
+      ncesSchoolId: null,
+      schoolName: '',
+      schoolDisplayName: null,
+      schoolState: '',
+      schoolZip: '',
+      schoolLocation: '',
+      showSchoolInfoErrors: false,
+      showSchoolInfoUnknownError: false,
+    });
```

Comment:
This appears to be the same as the initial state on line 79, plus the last 2 `showSchoolInfo*` fields. This raises a couple questions: 

 1. Should those 2 also be initialized on line 79?
 2. If the 2 state objects are identical, then why don't we wrap these into a single static property, e.g. `initialState` and set both the initial `state` and this `setState` to that value?

Reply:
Yes, those should be the same. I changed it to reference a common definition of that state.